### PR TITLE
data registry fixtures

### PR DESCRIPTION
**Pull request by snopoke**
## Summary
New fixture provider for data registries.

Spec: https://docs.google.com/document/d/1FCSxVBoBG8LnmogMwei1nb6eC74eUhs7HWuJmBg6i3k/edit#

## Feature Flag
DATA_REGISTRY

## Product Description
Provide list of data registries and granted domains in each registry as a fixture for use within applications.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
Added

### QA Plan
None

### Safety story
Change is isolated but could impact restore requests. Restore has good test coverage so the risk is low.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/registry/fixtures.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        if not _should_sync(restore_state):
+            return []
+
+        apps = _get_apps(restore_state)
+        registry_slugs = {
+            module.search_config.data_registry
+            for app in apps
+            for module in app.get_modules()
+            if module.search_config.data_registry
+        }
```
Comment (esoergel) [inline corehq/apps/registry/fixtures.py:25]: This'd be a good cache target - the result is tiny, fairly expensive to compute, and only varies by domain.  It'd be nice to cache the whole fixture xml, but I'm not sure we have a good framework for that, do we?

**Review by esoergel — APPROVED** [review]

File: corehq/apps/registry/fixtures.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/registry/fixtures.py:25]: the apps are cached so this shouldn't be that expensive should it?

We do have some caching that we use for lookup tables, I can take a look to see if we can use it at all and how we'd know when to bust the cache.

Comment (esoergel) [reply corehq/apps/registry/fixtures.py:25]: I don't know - I was mostly thinking about the overhead of wrapping large apps.  Not sure how much of a contributor it'd be in practice, but yeah we can always look at example restores once this is live and see.

Comment (snopoke) [reply corehq/apps/registry/fixtures.py:25]: https://dimagi-dev.atlassian.net/browse/USH-1278

Comment (snopoke) [reply corehq/apps/registry/fixtures.py:25]: > I don't know - I was mostly thinking about the overhead of wrapping large apps. Not sure how much of a contributor it'd be in practice, but yeah we can always look at example restores once this is live and see.

Not that this makes it correct but the UCR fixture does the same thing: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/fixtures/mobile_ucr.py#L122-L128

Comment (esoergel) [reply corehq/apps/registry/fixtures.py:25]: Oh good point - there was a recent ticket about that fixture being surprisingly slow even with trivially small reports - I'll see if I can dig it up.

**Pull request merged**